cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+  bool operator==(column_wrapper<ColumnType> const& rhs) const {
+    if (the_column.size != rhs.the_column.size) return false;
+    if (the_column.dtype != rhs.the_column.dtype) return false;
+    if (the_column.null_count != rhs.the_column.null_count) return false;
+    if (the_column.dtype_info.time_unit != rhs.the_column.dtype_info.time_unit)
+      return false;
+
+    if (!(the_column.data && rhs.the_column.data))
+      return false;  // if one is null but not both
+
+    if (not thrust::all_of(rmm::exec_policy()->on(0),
+                           thrust::make_counting_iterator(0),
+                           thrust::make_counting_iterator(the_column.size),
+                           elements_equal{the_column, rhs.the_column})) {
+      return false;
+    }
+
+    CUDA_RT_CALL(cudaPeekAtLastError());
+
+    return true;
+  }
+
+ private:
+  /**---------------------------------------------------------------------------*
+   * @brief Allocates and initializes the underyling gdf_column with host data.
+   *
+   * Creates a gdf_column and copies data from the host for it's data and
+   * bitmask. Sets the corresponding dtype based on the column_wrapper's
+   * ColumnType.
+   *
+   * @param host_data The vector of host data to copy to device for the column's
+   * data
+   * @param host_bitmask Optional vector of host data for the column's bitmask
+   * to copy to device
+   *---------------------------------------------------------------------------**/
+  void initialize_with_host_data(
+      std::vector<ColumnType> const& host_data,
+      std::vector<gdf_valid_type> const& host_bitmask =
+          std::vector<gdf_valid_type>{}) {
+    // thrust::device_vector takes care of host to device copy assignment
+    data = host_data;
+
+    // Fill the gdf_column members
+    the_column.data = data.data().get();
+    the_column.size = data.size();
+    the_column.dtype = cudf::gdf_dtype_of<ColumnType>();
+    gdf_dtype_extra_info extra_info;
+    extra_info.time_unit = TIME_UNIT_NONE;
+    the_column.dtype_info = extra_info;
+
+    // If a validity bitmask vector was passed in, allocate device storage
+    // and copy its contents from the host vector
+    if (host_bitmask.size() > 0) {
+      gdf_size_type const required_bitmask_size{
+          gdf_get_num_chars_bitmask(host_data.size())};
+
+      if (host_bitmask.size() < static_cast<size_t>(required_bitmask_size)) {
+        throw std::runtime_error("Insufficiently sized bitmask vector.");
+      }
+      bitmask = host_bitmask;
+      the_column.valid = bitmask.data().get();
+    } else {
+      the_column.valid = nullptr;
+    }
+
+    gdf_error result = set_null_count(&the_column);
+    if (GDF_SUCCESS != result) {
+      throw std::runtime_error("Failed to set null count. Error code: " +
+                               std::to_string(result));
+    }
+  }
+
+  rmm::device_vector<ColumnType> data;
+  rmm::device_vector<gdf_valid_type> bitmask;
```

Comment:
Another thing that needs unification is the naming for these things.

bitmask?
stencil?
valid?
null?
null indicators?
validity?
nullness?

Reply:
Add it to the list of things we need in a coding style guide :smile: